BankDemo.py

```diff
@@ -1,129 +1,160 @@
 """"importing modules"""
-from flask import Flask, redirect, url_for, jsonify
-import pyodbc
+import  json
+from flask import Flask, redirect, url_for, abort, request
+import DBconnection
+import DBqueries
 
 
-app = Flask(__name__)
+# Calling Flask API
+APP = Flask(__name__)
 
 
-def init_db():
-    """"connecting to database"""
-    server = '192.168.18.36'
-    db = 'HBK_Test'
-    user = 'kailashp'
-    password = 'Welcome@731'
-    con = pyodbc.connect('Driver={ODBC Driver 17 for SQL Server};'
-                         'SERVER=' + server + ';DATABASE=' + db +
-                         ';UID=' + user + ';PWD=' + password)
-    cursor = con.cursor()
-    return cursor
+CURSOR = DBconnection.init_db()
 
 
-cursor = init_db()
-
-
-@app.route('/login/<userID>,<password>')
+@APP.route('/login/<userID>,<password>', methods=['GET', 'POST'])
 def login(userID, password):
     """"checking login details"""
-    login_object = get_login_details(userID, password)
-    if type(login_object) is pyodbc.Row:
-        return redirect(url_for('accountdetails', userid=userID))
-    return "UserID or password is invalid"
+    try:
+        login_object = get_login_details(userID, password)
+        if login_object:
+            return redirect(url_for('accountdetails', userid=userID, password=password))
+        return redirect(url_for('add_user'), code=307)
+        # abort(500)
+    except Exception as e:
+        print(e)
 
 
-def get_login_details(userID, pswrd):
+def get_login_details(userid, pswrd):
     """"getting login details"""
-    cursor.execute("SELECT userID, "
-                   " Password from bankuser_details"
-                   " where userID = ? and Password = ?",
-                   userID, pswrd)
-    rows = cursor.fetchall()
-    if rows:
-        for row in rows:
-            return row
-
-
-@app.route('/accountdetails/<userid>')
-def accountdetails(userid):
-    """" displaying account details"""
-    accnt_detls_object = get_accnt_details(userid)
-    if type(accnt_detls_object) is pyodbc.Row:
-        return jsonify({"accountnumber": accnt_detls_object[0],
-                        "accounttype": accnt_detls_object[2],
-                        "balance": accnt_detls_object[3],
-                        "branch": accnt_detls_object[4]})
-    return "UserID or password is invalid"
-
-
-def get_accnt_details(userid):
-    """getting userID details from database"""
-    cursor.execute("SELECT * from bankaccount_details"
-                   " where userID = ?", userid)
-    rows = cursor.fetchall()
-    if rows:
-        for row in rows:
-            return row
-
-
-@app.route('/transactiondetails/<accountnum>')
+    CURSOR.execute(DBqueries.login_query(), userid, pswrd)
+    rows = CURSOR.fetchall()
+    for row in rows:
+        return row
```

Comment:
Why there is a for loop if you want to return something just after first iteration.